k8s/apis/k8s/v1alpha1/vineyardd_types.go

```diff
@@ -0,0 +1,193 @@
+	EtcdCmd string `json:"etcdCmd,omitempty"`
+	// endpoint of etcd
+	// +kubebuilder:validation:Optional
+	EtcdEndpoint string `json:"etcdEndpoint,omitempty"`
+	// path prefix in etcd
+	// +kubebuilder:validation:Optional
+	EtcdPrefix string `json:"etcdPrefix,omitempty"`
+	// enable the metrics in vineyardd
+	// +kubebuilder:validation:Optional
+	EnableMetrics bool `json:"enableMetrics,omitempty"`
+	// whether to print metrics for prometheus
+	// +kubebuilder:validation:Optional
+	EnablePrometheus bool `json:"enablePrometheus,omitempty"`
+	// ipc socket file location
+	// +kubebuilder:validation:Optional
+	Socket string `json:"socket,omitempty"`
+	// the configuration of spilling
+	// +kubebuilder:validation:Optional
+	SpillConfig SpillConfig `json:"spillConfig,omitempty"`
+	// the configuration of serialization
+	// +kubebuilder:validation:Optional
+	SerializeConfig SerializeConfig `json:"serializeConfig,omitempty"`
+	// memory threshold of streams (percentage of total memory)
+	// +kubebuilder:validation:Optional
+	StreamThreshold int64 `json:"streamThreshold,omitempty"`
+	// shared memory size for vineyardd
+	// +kubebuilder:validation:Optional
+	SharedMemorySize string `json:"sharedMemorySize,omitempty"`
+	// synchronize CRDs when persisting objects
+	// +kubebuilder:validation:Optional
+	SyncCRDs bool `json:"syncCRDs,omitempty"`
+}
+
+// SpillConfig holds all configuration about spilling
+type SpillConfig struct {
+	// the name of the spill config
+	// +kubebuilder:validation:Optional
+	Name string `json:"name,omitempty"`
+	// the path of spilling
+	// +kubebuilder:validation:Optional
+	Path string `json:"path,omitempty"`
+	// low watermark of spilling memory
+	// +kubebuilder:validation:Optional
+	SpillLowerRate string `json:"spillLowerRate,omitempty"`
+	// high watermark of triggering spiling
+	// +kubebuilder:validation:Optional
+	SpillUpperRate string `json:"spillUpperRate,omitempty"`
+	// the PersistentVolumeSpec of the spilling PV
+	// +kubebuilder:validation:Optional
+	PersistentVolumeSpec corev1.PersistentVolumeSpec `json:"persistentVolumeSpec,omitempty"`
+	// the PersistentVolumeClaimSpec of the spill file
+	// +kubebuilder:validation:Optional
+	PersistentVolumeClaimSpec corev1.PersistentVolumeClaimSpec `json:"persistentVolumeClaimSpec,omitempty"`
+}
+
+// SerializeConfig holds all configuration about serializetion
+type SerializeConfig struct {
+	// the name of the serialization config
+	// +kubebuilder:validation:Optional
+	Name string `json:"name,omitempty"`
+	// the path of serializetion
+	// +kubebuilder:validation:Optional
+	Path string `json:"path,omitempty"`
+	// the configuration of the serialization PV
+	// +kubebuilder:validation:Optional
+	PersistentVolumeSpec corev1.PersistentVolumeSpec `json:"persistentVolumeSpec,omitempty"`
+	// the PersistentVolumeClaimSpec of the serializtion
+	// +kubebuilder:validation:Optional
+	PersistentVolumeClaimSpec corev1.PersistentVolumeClaimSpec `json:"persistentVolumeClaimSpec,omitempty"`
```

Comment:
If I understand it correctly, `serialization` is a user-triggerred operator (e.g., invoke `vineyard.io.serialize()` in user's workload).

Why do we need add such a configuration to the deployment of vineyardd itself?

The serialization volume should be mounted along with the workload, rather than vineyardd.

Reply:
Make sense to me, what I imagined earlier is that both vineyardd and user pods need to recognize this `serialization 
 path` at the same time. I will delete the field later.